content/tutorials/git_authentication/index.Rmd

```diff
@@ -0,0 +1,236 @@
+
+Most of the material in this tutorial is explained in length in https://happygitwithr.com/connect-intro.html.
+The idea of this tutorial is to condense some of this material and make clear recommendations for use at INBO. 
+We will mainly follow [the advice given in the usethis package](https://usethis.r-lib.org/articles/articles/git-credentials.html).
+
+
+## TL;DR
+
+Use the following commands to configure some global git options (if you haven't done this already):
+
+```{r eval=FALSE}
+gert::git_config_global_set("user.name", "Your Name") # change Your Name
+gert::git_config_global_set("user.email", "your@email.com") # change your@email.com
+gert::git_config_global_set("init.defaultBranch", "main")
+```
+
+Use the following commands to create a Personal Access Token (PAT) and add it to the Git Credential Manager: 
+
+```{r eval=FALSE}
+usethis::git_vaccinate()
+usethis::create_github_token() #browser opens, follow instructions
+gitcreds::gitcreds_set() #paste PAT
+```
+
+Check if everything is OK:
+
+```{r eval=FALSE}
+usethis::git_sitrep()
+```
+
+
+## Modes of authentication
+
+There are various ways to authenticate and these modes depend on how you access your resources in GitHub.
+Here are ways of authentication that we recommend for various ways of accessing your GitHub resources.
+
+- Access to [GitHub](https://github.com/) via the browser
+    - Username and password combined with two-factor authentication
+- Interact with [GitHub as a Git Server](https://github.com/) via RStudio _and_ Access to GitHub via the [GitHub REST API](https://docs.github.com/en/rest)
+    - The HTTPS protocol combined with a Personal Access Token (PAT) as credential (preferably cached in the Git Credential Manager)
+- Access to Github via the [GitHub Command Line Interface](https://cli.github.com/)
+    - See [gh auth login](https://cli.github.com/manual/gh_auth_login), which takes you to an interactive authentication via your browser. This procedure can also be used in a non-interactive way when authentication is needed inside a _GitHub Actions workflow_ (not discussed further in this tutorial).
+
+As can be seen from the above overview, authentication is necessary in lots of instances.
+This ranges from something simple such as managing your repositories online to more complex interactions.
+
+We fully agree with the [recommendations given in the usethis package](https://usethis.r-lib.org/articles/articles/git-credentials.html) and repeat these here:
+
+1. Turn on two-factor authentication for your GitHub account.
+1. Adopt HTTPS as your Git transport protocol.
+1. Use a personal access token (PAT) for all Git remote operations from the command line or from R.
+1. Allow tools to store and retrieve your credentials from the Git Credential Manager.
+
+This deserves some further explanation.
+The first recommendation essentially is an extra layer of security compared to a simple username and password only authentication.
```

Comment:
@peterdesmet can't we enforce 2FA at the organisation level?

Reply:
It's possible. But at this point it would be more interesting to know how many people have 2FA enabled already. One of the INBO organisation owners/administrators [can check this](https://docs.github.com/en/enterprise-server@3.4/organizations/keeping-your-organization-secure/managing-two-factor-authentication-for-your-organization/viewing-whether-users-in-your-organization-have-2fa-enabled)